Makefile

```diff
@@ -399,7 +382,12 @@ SRC = $(wildcard src/*/*/*/*.cc src/*/*/*.cc src/*/*.cc src/*.cc)
 ifneq ($(USE_NGRAPH),1)
     SRC := $(foreach f,$(SRC),$(if $(findstring src/ngraph,$f),,$f))
 endif
+
 OBJ = $(patsubst %.cc, build/%.o, $(SRC))
+ifeq ($(USE_NGRAPH),1)
+  $(OBJ): ngraph
```

Comment:
I'm not familiar with this syntax, what does this line do?

Reply:
`$(OBJ)` expands to a list of `.o` files.  This states that every one of those `.o` files is dependent on the `ngraph` target first being built.

This was the cleanest way I could think of to ensure that ngraph was fully built and installed before any of the code that uses it.